### DISCUSSION TITLE
Parallel make or any other way to boost compilation?

### DISCUSSION THREAD

**Discussion by shoce** [libvips/libvips#2152] [Q&A] [answered]
Is there any faster way to build libvips than ordinary `make`?

**Comment by jcupitt** [comment, marked as answer]
Hi @shoce,

Sure, use `-j N` to do an N-way parallel build. For example, I see:

```
$ time make -j 32 >& log
real	0m17.595s
user	2m47.918s
sys	0m33.571s
```

**Reply by kmartinez** [reply]
the trick is usually to check how many cores (incl hyperthreads) you have and set that as -j value. Sometimes adding 1 gets you a small speed-up too. If you make a lot of different projects its worth making an alias in .bashrc like
alias makep="make -j 5"

**Reply by shoce** [reply]
Yes, thanks a lot, i ended up with `make --jobs $(nproc)`.
